Assets/Scripts/Foliage/SpawnFoliage.cs

```diff
@@ -1,161 +1,247 @@
+
+    [Header("Misc")]
+    [SerializeField] private int seed = 0;
+    [SerializeField] private bool DEBUG = false;
+
+
+    private int treeIndex = 0;
+    private int treeSpawnIndex = 0;
+    private Vector3[] treePositions = new Vector3[prefabLimit];
+
+    private int bushIndex = 0;
+    private int bushSpawnIndex = 0;
+    private Vector3[] bushPositions = new Vector3[prefabLimit];
+
+    private int stoneIndex = 0;
+    private int stoneSpawnIndex = 0;
+    private Vector3[] stonePositions = new Vector3[prefabLimit];
+
+
+    // Private members
+    private GameObject foliageHandler;
+
+    private Vector3 planetCenter;
+    private Planet planet;
+    private float planetRadius;
+    private float waterLevel;
+
+    void Update()
     {
-        Vector3 position;
-        RaycastHit hit;
 
-        // Loops thought all tree locations
-        for (int i = 0; i < treeLimit; i++)
+        if(planet != null && foliageHandler != null)
         {
-            // Creates a ball around the planet which shoots rays to check if it's legal to spawn foliage
-            position = Random.onUnitSphere * rayBallRadius;
-
-            if (Physics.Raycast(position, -position, out hit, rayLenght))
+            GameObject[] cameras = GameObject.FindGameObjectsWithTag("MainCamera");
+            for(int i = 0; i < cameras.Length; i++)
             {
-                if (hit.transform.tag == "Foliage" ||                         // Checks if it doesnt hit ground
-                    Mathf.Abs(Vector3.Angle(position, hit.normal)) > 30 ||    // Checks the angle
-                    hit.distance < (rayBallRadius - planetRadius) - treeLine)        // Checks the height
+                // Tries to spawn 100 of each every frame we are near the planet
+                if ((cameras[i].transform.position - planet.transform.position).magnitude < 3000)
+                {
+
+                    for (int j = 100; j > 0; j--)
+                    {
+                        if (treeIndex > treeSpawnIndex) plantTree();
+                        else break;
+                    }
+                    for (int j = 100; j > 0; j--)
+                    {
+                        if (bushIndex > bushSpawnIndex) plantBush();
+                        else break;
+                    }
+                    for (int j = 100; j > 0; j--)
+                    {
+                        if (stoneIndex > stoneSpawnIndex) plantStone();
+                        else break;
+                    }
+
+                }
+                // Delets all foliage when leaving
+                else if((cameras[i].transform.position - planet.transform.position).magnitude > 5000)
                 {
-                    // Illegal spawn point
-                    i--;
-                    continue;
+                    if (foliageHandler.transform.childCount > 0)
+                    {
+
+                        Destroy(foliageHandler);
```

Comment:
Not really something you have to change right now, but we should perform tests on Instantiating/Destroying vs Enable/Disable. Both have their disadvantages and advantages and we should check what aspects our project needs to focus on.
Here is one discussion about it:
https://forum.unity.com/threads/instantiate-destroy-vs-enable-disable.289507/

Reply:
Will look into this in the future.